app/models/spree_avatax_certified/line.rb

```diff
@@ -74,6 +74,22 @@ def shipment_lines_array
       ship_lines
     end
 
+    def shipping_refund_lines_array
+      @logger.info('build shipment refund lines')
+
+      ship_refund_lines = []
+      order.shipments.each do |shipment|
+
+        if shipment.tax_category
```

Comment:
Does shipment have a tax_category field?

Reply:
It does but we don't used it. `shipping_refund_lines_array` is overridden in hw_admin

Reply:
I didn't want to bring the concept of master shipment into the gem. It's so ugly this thing :(